portfolio/src/main/java/com/google/sps/data/FullBook.java

```diff
@@ -0,0 +1,171 @@
+    switch (fieldName) {
+      case "title": {
+        return this.title;
+      }
+      case "genre": {
+        return this.title;
+      }
+      case "language": {
+        return this.language;
+      }
+      case "description": {
+        return this.description;
+      }
+      case "infoLink": {
+        return this.infoLink;
+      }
+      case "pageCount": {
+        return this.pageCount;
+      }
+      case "publishedDate": {
+        return this.publishedDate;
+      }
+      case "publisher": {
+        return this.publisher;
+      }
+      case "maturityRating": {
+        return this.maturityRating;
+      }
+      case "thumbnail": {
+        return this.thumbnail;
+      }
+      case "isbn": {
+        return this.isbn;
+      }
+      default: {
+        throw new NoSuchFieldException();
+      }
+    }
+  }
+
+  public void setStringField(String fieldName, String value) throws NoSuchFieldException {
+    switch (fieldName) {
+      case "title":
+        this.title = value;
+        break;
+      case "language":
+        this.language = value;
+        break;
+      case "genre":
+        this.genre = value;
+        break;
+      case "description":
+        this.description = value;
+        break;
+      case "infoLink":
+        this.infoLink = value;
+        break;
+      case "pageCount":
+        this.pageCount = value;
+        break;
+      case "publishedDate":
+        this.publishedDate = value;
+        break;
+      case "publisher":
+        this.publisher = value;
+        break;
+      case "maturityRating":
+        this.maturityRating = value;
+        break;
+      case "thumbnail":
+        this.thumbnail = value;
+        break;
+      case "isbn":
+        this.isbn = value;
+        break;
+      default: {
+        throw new NoSuchFieldException();
+      }
+    }
+  }
+
+  public ArrayList<String> getArrayField(String fieldName) throws NoSuchFieldException {
+    if (fieldName.equals("categories")) {
+      return this.categories;
+    } else {
+      return this.authors;
+    }
+  }
+
+  public void setArrayField(String fieldName, List<String> value) {
+    if (fieldName.equals("categories")) {
+      this.categories = (ArrayList<String>) value;
+    } else {
+      this.authors = (ArrayList<String>) value;
+    }
+  }
+
+  public FullBook() {}
+
+  @Override
+  public String toString() {
+    return String.format("Title: %s with isbn: %s", this.title, this.isbn);
+  }
+
+  public String toStringAll() {
+    String format = "";
+    format += String.format("Title: %s with genre:%s in %s with desc: %s", this.title, this.genre,
+        this.language, this.description);
+    format +=
+        String.format("This book has %s pages, published on %s by %s with a maturity rating of %s",
+            this.pageCount, this.publishedDate, this.publisher, this.maturityRating);
+    format += String.format("infoLink %s and thumbnail %s", this.infoLink, this.thumbnail);
```

Comment:
for later: since the thumbnail is simply a string at this point, maybe we could use [Java](https://stackoverflow.com/questions/10479621/how-to-print-image-in-java) or the html/js files to actually draw the thumbnail too?